src/display/transport_stream.js

```diff
@@ -0,0 +1,241 @@
+/* Copyright 2012 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { assert, createPromiseCapability } from '../shared/util';
+
+ /** @implements {IPDFStream} */
+var PDFDataTransportStream = (function PDFDataTransportStreamClosure() {
+  function PDFDataTransportStream(params, pdfDataRangeTransport) {
+    this.transportReadyCapability = createPromiseCapability();
```

Comment:
we don't need transportReadyCapability, since we moved stream to the main thread and PDFDataTransportStream can cache data as soon as addProgressiveReadListener bound. If I'm not mistaken, we can safely remove this guard.